### DISCUSSION TITLE
How to implement CMFD on continuous FEM

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#27643] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi, everyone.

I want to implement CMFD (Coarse Mesh Finite Difference method) on CFEM (Continuous Finite Element Method), i.e. the transport calculations are based on the CFEM method. However, there are shared Nodes between different elements of the CFEM mesh, so when accelerating the transport fluxes based on the results of the coarse mesh, there is a question of which Elem the Node should be based on to do the scaling of its fluxes? For example, a Node may be shared by four Elems, and their scaling ratios are 1, 2, 3, and 4 respectively, how should the flux scaling ratio of this Node be determined?

Thank you for your attention. Hopefully someone can give advice.

**Comment by GiudGiud** [comment]
Hello

I would suggest to blend the 4 ratios together.
At convergence, all scaling ratios will be 1 so it wont matter. 

Guillaume

**Reply by hityyds** [reply]
Thank you for your response. I tried using the average ratio, but it still didn't work. I tested it, and the program runs correctly if I add the DGKernel, but it fails when I remove it. I think this might be because using the average ratio disrupts the local consistency between the coarse and fine meshes. After this processing, the flux at this node doesn't reach its ideal value for all four elements.

**Reply by hityyds** [reply]
From this perspective, CFEM and CMFD are incompatible. I'm ready to give up. Thank you again for your advice.

**Comment by lindsayad** [comment]
@yaqiwang @jthano do you guys do CMFD with CFEM?

**Reply by hityyds** [reply]
Thank you for your attention, but I'm ready to give up. I have tried using the average ratio, but it still didn't work. I tested it, and the program runs correctly if I add the DGKernel, but it fails when I remove it. I think this might be because using the average ratio disrupts the local consistency between the coarse and fine meshes. After this processing, the flux at this node doesn't reach its ideal value for all four elements. From this perspective, CFEM and CMFD are incompatible.

**Reply by YaqiWang** [reply, marked as answer]
No. It is possible (https://www.tandfonline.com/doi/abs/10.13182/NSE12-50). We typically do fine mesh diffusion acceleration with CFEM (https://www.tandfonline.com/doi/abs/10.13182/NSE12-83). We do CMFD typically with DFEM (https://doi.org/10.1016/j.anucene.2024.110365).

**Reply by hityyds** [reply]
Thank you for the reminder. I will carefully study the content of these articles.
